src/modules/subscription/utils.ts

```diff
@@ -0,0 +1,91 @@
+import { NotificationType, SubscriptionDetails } from '@dcl/schemas'
+import { SubscriptionDetailsCamelCase, SubscriptionGroupKeys, SubscriptionGroups } from './types'
+
+export function toCamelCase(str: string): string {
+  return str.replace(/([-_][a-z])/g, group => group.toUpperCase().replace('-', '').replace('_', ''))
+}
+
+function transformKeysToCamelCase<T extends Record<string, unknown>>(inputObject: T): Record<string, unknown> {
+  if (typeof inputObject !== 'object' || inputObject === null) {
+    return inputObject
+  }
+
+  if (Array.isArray(inputObject)) {
+    return inputObject.map(item => transformKeysToCamelCase(item as Record<string, unknown>)) as unknown as Record<string, unknown>
```

Comment:
You seem to be casting the item type to a `Record<string, unknown>` here to make it work, but item can be a primitive. It seems that the type of the function is not correct, as it should `receive T | unknown` or any other signature that allows the recursive processing of any type.